oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/JwtProcessors.java

```diff
@@ -235,4 +249,52 @@ public PublicKeyJwtProcessorBuilder jwsAlgorithm(String jwsAlgorithm) {
 			return jwtProcessor;
 		}
 	}
+
+	/**
+	 * A builder for creating a Nimbus {@link JWTProcessor} using a {@code SecretKey}.
+	 */
+	public static final class SecretKeyJwtProcessorBuilder {
+		private final SecretKey secretKey;
+		private JWSAlgorithm jwsAlgorithm = JWSAlgorithm.HS256;
+
+		private SecretKeyJwtProcessorBuilder(SecretKey secretKey) {
+			Assert.notNull(secretKey, "secretKey cannot be null");
+			this.secretKey = secretKey;
+		}
+
+		/**
+		 * Use the given
+		 * <a href="https://tools.ietf.org/html/rfc7515#section-4.1.1" target="_blank">algorithm</a>
+		 * when generating the MAC.
+		 *
+		 * The value should be one of
+		 * <a href="https://tools.ietf.org/html/rfc7518#section-3.2" target="_blank">HS256, HS384 or HS512</a>.
+		 *
+		 * @param jwsAlgorithm the JWS algorithm to use
+		 * @return a {@link SecretKeyJwtProcessorBuilder} for further configurations
+		 */
+		public SecretKeyJwtProcessorBuilder jwsAlgorithm(MacAlgorithm jwsAlgorithm) {
```

Comment:
Does it make sense to rename this method and argument now that it is of type MacAlgorithm?